numpy/_core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -510,7 +510,7 @@ create_custom_field_dtype(PyObject *NPY_UNUSED(mod), PyObject *args)
     if (dtype->type_num != NPY_VOID || dtype->fields == NULL ||
             !PyDict_CheckExact(dtype->fields) ||
             PyTuple_Size(dtype->names) != 1 ||
-            !PyDataType_REFCHK(dtype) ||
+            !PyDataType_REFCHK((PyArray_Descr *)dtype) ||
```

Comment:
Basically feel one should avoid recasting a legacy type into a "real" one (but really am nitpicking here)

Reply:
That gets annoying since you cannot *assign* using the macros.  Having both around is reasonable.
Also, what you don't see is that I removed a lot of changed lines of code again by going the cast route.

It would be nice to just define some of these internally in a way that supports the subclass `LegacyDescr *`, but it was few enough instances where this was needed that I didn't think it was very important.